### PR TITLE
Migrate 'components/sidebar_right/github_items.js' to TypeScript

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
I worked on this migrate a couple of month ago and finally got the time to push the PR. It's an example migration for a plugin component to typescript. I shied away also migrating `index.jsx` as the would require typing the global state.

#### Ticket Link
None

### CODE REVIEW

**Comment by hanzei** [comment]
@larkox @mickmister Take your time with reviewing this PR. There is no time pressure.

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 1]
```diff
@@ -1,47 +1,121 @@
...
-import CrossIcon from 'images/icons/cross.jsx';
-import DotIcon from 'images/icons/dot.jsx';
-import TickIcon from 'images/icons/tick.jsx';
-import SignIcon from 'images/icons/sign.jsx';
-import ChangesRequestedIcon from 'images/icons/changes_requested.jsx';
+import CrossIcon from '../../images/icons/cross';
+import DotIcon from '../../images/icons/dot';
+import TickIcon from '../../images/icons/tick';
+import SignIcon from '../../images/icons/sign';
+import ChangesRequestedIcon from '../../images/icons/changes_requested';
```
Comment (hanzei) [inline webapp/src/components/sidebar_right/github_items.tsx:19]: I wasn't able to get webpack to pickup the modules without a relative path.

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 2]
```diff
@@ -1,47 +1,121 @@
...
+
+interface GithubItemsProps {
+    items: Item[];
+    theme: Theme;
+}
+
+function GithubItems(props: GithubItemsProps) {
     const style = getStyle(props.theme);
 
+    //console.log('props.items:' + props.items);
```
Comment (larkox) [inline webapp/src/components/sidebar_right/github_items.tsx:94]: Should this be removed?

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 3]
```diff
@@ -56,6 +130,14 @@ function GithubItems(props) {
                 </strong>);
         }
 
+        let titelText = '';
```
Comment (larkox) [inline webapp/src/components/sidebar_right/github_items.tsx:133]: Typo?

**Review by larkox — APPROVED** [review]

File: webapp/src/components/sidebar_right/github_items.tsx (hunk 2 above)
Comment (hanzei) [reply webapp/src/components/sidebar_right/github_items.tsx]: Good catch :+1:

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 4]
```diff
@@ -1,47 +1,120 @@
...
+interface User {
+    login: string;
+}
+
+interface Review {
+    state: string;
+    user: User;
+}
+
+interface Item {
```
Comment (mickmister) [inline webapp/src/components/sidebar_right/github_items.tsx:51]: Can we have these defined in a file dedicated to types?

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 5]
```diff
@@ -1,47 +1,120 @@
...
         if (item.number) {
-            const iconProps = {
+            const iconProps: IconProps = {
                 size: 'small',
                 verticalAlign: 'text-bottom',
             };
 
             let icon;
-            if (item.pull_request) {
+            if (item.pullRequest) {
```
Comment (mickmister) [inline webapp/src/components/sidebar_right/github_items.tsx:117]: Was this broken before?

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 6]
```diff
@@ -1,47 +1,120 @@
...
+    labels?: Label[];
+
+    // PRs
+    status?: string;
+    mergeable?: boolean;
+    requestedReviewers?: string[];
+    reviews?: Review[];
+
+    // Assignments
+    pullRequest?: undefined;
```
Comment (mickmister) [inline webapp/src/components/sidebar_right/github_items.tsx:77]: I think `unknown` would be a better type for this

```suggestion
    pullRequest?: unknown;
```

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 7]
```diff
@@ -56,6 +129,14 @@ function GithubItems(props) {
...
         }
 
+        let titleText = '';
+        if (item.title) {
+            titleText = item.title;
+        } else if (item.subject?.title) {
+            titleText = item.subject.title;
+        }
+
+        let title:JSX.Element | null = null;
```
Comment (mickmister) [inline webapp/src/components/sidebar_right/github_items.tsx:139]: 0/5, then you can remove the `else` below

```suggestion
        let title: React.ReactNode = titleText;
```

**Review by mickmister — APPROVED** [review]
A few comments, nothing blocking. LGTM :+1:

File: webapp/src/components/sidebar_right/github_items.tsx (hunk 5 above)
Comment (hanzei) [reply webapp/src/components/sidebar_right/github_items.tsx:117]: I think the JSON parser accepts both fields names. So no, it was not broken.

File: webapp/src/components/sidebar_right/github_items.tsx (hunk 4 above)
Comment (hanzei) [reply webapp/src/components/sidebar_right/github_items.tsx:51]: 2/5 that it's not worth have a separate file if the types are only used in one file.

**Comment by hanzei** [comment]
@DHaussermann Would you be fine testing this as part of the release process?

**Review by DHaussermann — APPROVED** [review]
This can be tested post merge.
The regression testing needed will be covered as part of the release testing process. Please merge.

LGTM!

**Pull request merged**